src/Model.php

```diff
@@ -1944,3 +1924,24 @@ public function __debugInfo(): array
 
     // }}}
 }
+
+class RecordNotFoundException extends Exception
```

Comment:
This should probably be defined in separate file.

Reply:
yes

Reply:
There is one more class `ArrayCallbackIterator` in the `IteratorQuery` file.
Move out?

Reply:
yes, always, max. 1 class per file

Reply:
The idea is that these classes are used withing the autoloaded class or its children.

But if it is more clear then I will separate.

Reply:
No assumptions when they are loaded, never, I think it is also enforced by some PSR.